kinto/core/resource/__init__.py

```diff
@@ -1087,7 +1087,9 @@ def _build_pagination_token(self, sorting, last_record, offset):
         }
 
         for field, _ in sorting:
-            token['last_record'][field] = last_record[field]
+            last_value = find_nested_value(last_record, field)
+            if last_value is not None:
+                token['last_record'][field] = last_value
```

Comment:
Also note that without a JSON schema to validate provided parameters against, so all fields are then optional by design, the last record from a chunk might or might not have the field matching the provided key set. For which case I don't have any solution in mind, sadly.